sycl/include/syclcompat/math.hpp

```diff
@@ -208,6 +241,151 @@ inline unsigned vectorized_isgreater<sycl::ushort2, unsigned>(unsigned a,
+  return sycl::fmin<std::common_type_t<T1, T2>>(a, b);
+}
+sycl::half min(sycl::half a, sycl::half b) {
+  return sycl::fmin<sycl::half>(a, b);
+}
+
+template <typename T1, typename T2>
+std::enable_if_t<std::is_integral_v<T1> && std::is_integral_v<T2>,
+                 std::common_type_t<T1, T2>>
+max(T1 a, T2 b) {
+  return sycl::max<std::common_type_t<T1, T2>>(a, b);
+}
+template <typename T1, typename T2>
+std::enable_if_t<std::is_floating_point_v<T1> && std::is_floating_point_v<T2>,
+                 std::common_type_t<T1, T2>>
+max(T1 a, T2 b) {
+  return sycl::fmax<std::common_type_t<T1, T2>>(a, b);
+}
+sycl::half max(sycl::half a, sycl::half b) {
+  return sycl::fmax<sycl::half>(a, b);
+}
+
+/// Performs 2 elements comparison and returns the bigger one. If either of
+/// inputs is NaN, then return NaN.
+/// \param [in] a The first value
+/// \param [in] b The second value
+/// \returns the bigger value
+template <typename ValueT, typename ValueU>
+inline std::common_type_t<ValueT, ValueU> fmax_nan(const ValueT a,
+                                                   const ValueU b) {
+  if (detail::isnan(a) || detail::isnan(b))
+    return NAN;
+  return sycl::fmax<std::common_type_t<ValueT, ValueU>>(a, b);
+}
+template <typename ValueT, typename ValueU>
+inline sycl::vec<std::common_type_t<ValueT, ValueU>, 2>
+fmax_nan(const sycl::vec<ValueT, 2> a, const sycl::vec<ValueU, 2> b) {
+  return {fmax_nan(a[0], b[0]), fmax_nan(a[1], b[1])};
+}
+
+/// Performs 2 elements comparison and returns the smaller one. If either of
+/// inputs is NaN, then return NaN.
+/// \param [in] a The first value
+/// \param [in] b The second value
+/// \returns the smaller value
+template <typename ValueT, typename ValueU>
+inline std::common_type_t<ValueT, ValueU> fmin_nan(const ValueT a,
+                                                   const ValueU b) {
+  if (detail::isnan(a) || detail::isnan(b))
+    return NAN;
+  return sycl::fmin<std::common_type_t<ValueT, ValueU>>(a, b);
+}
+template <typename ValueT, typename ValueU>
+inline sycl::vec<std::common_type_t<ValueT, ValueU>, 2>
+fmin_nan(const sycl::vec<ValueT, 2> a, const sycl::vec<ValueU, 2> b) {
+  return {fmin_nan(a[0], b[0]), fmin_nan(a[1], b[1])};
+}
+
+// pow functions overload.
+inline float pow(const float a, const int b) { return sycl::pown(a, b); }
+inline double pow(const double a, const int b) { return sycl::pown(a, b); }
+
+template <typename ValueT, typename ValueU>
+inline typename std::enable_if_t<std::is_floating_point_v<ValueT>, ValueT>
+pow(const ValueT a, const ValueU b) {
+  return sycl::pow(a, static_cast<ValueT>(b));
+}
+
+// FIXME: non-floating point values default to double, requires fp64. Affects
+// testing and documentation as well.
+template <typename ValueT, typename ValueU>
+inline typename std::enable_if_t<!std::is_floating_point_v<ValueT>, double>
+pow(const ValueT a, const ValueU b) {
+  return sycl::pow(static_cast<double>(a), static_cast<double>(b));
+}
```

Comment:
No. This was detected thanks to the tests we added during the migration process and was tagged as this may create potential problems, but it is mainly a design decision to be made.

The SYCL spec does not contemplate calling the `pow` function with non-floating point values, so this code covers those cases. The problem with its current implementation is: if a user tries to call `pow` with an integral `base`, it will try to cast that value by default to a `double`, causing errors in devices without `aspect::fp64`.

The context of the comment is to have a reminder and decide what to do with this functionality. Maybe instead of a FIXME a TODO would be more appropriate. What do you think?

Reply:
Ya that would be a bit clearer in terms of what actually needs to change

Reply:
Changed the FIXME for a TODO. Thanks for the feedback